### DISCUSSION TITLE
Overall Suggestions

### DISCUSSION THREAD

**Discussion by Bluemandan** [oxdc/obsidian-vertical-tabs#78] [Ideas]
I have been looking for group tabs for a bit now and I some how actually stumbled into your plugin, which works great! Here are some suggestions for possible features or additions to vertical tabs. From my POV **[High]** priority to **[Low]** Priority:
- [HIGH] Add some keywords into your description for the plugin, for more people to find the plugin. Such as "Tabs Organizing" or "Tab Groups"
- [High] Allow users to name grouped tab sections
- [Low] Add a custom sorting (Drag and Drop)

As well any of the [Google Tab Groups](https://files.catbox.moe/ahrlpc.mp4) functionality: 
- [MID] Color selector for tabs & Icons (stickers/svg)
  - [Low] Top Bar reflects all appearance changes made
- [High] Ability to make groups of tabs, that are not open, but will be open when you click to open the group.

Again love the plugin idea and everything works great!

**Comment by oxdc** [comment]
Hi! Thank you for your suggestions.

> [HIGH] Add some keywords into your description for the plugin, for more people to find the plugin. Such as "Tabs Organizing" or "Tab Groups"

Great idea!

> [High] Allow users to name grouped tab sections

This feature is under active development and will be shipped in the next major release.

> [Low] Add a custom sorting (Drag and Drop)

Custom sorting, including drag/drop and automatically sorting of *tabs*, is available now. I'm working on the same feature for *groups*. This requires some refactoring and will be available soon.

> [MID] Color selector for tabs & Icons (stickers/svg)

Vertical Tabs, ultimately, will manage and remember all the layouts, customizations, history, cursor positions, etc., and even sync them across devices securely in real time. Every customization feature is embedded into this framework. I'm designing the entire system right now. After that, these customizations will be easily integrated.

> [Low] Top Bar reflects all appearance changes made

Could you please explain this in more details? ~~By the way, I have trouble opening the video you posted. Is it possible to upload it here? Theoretically, you could upload it by just dragging and dropping the file into the text input. If it doesn't work, could you please upload it somewhere else? Like Google Drive or DropBox? Thanks.~~

> [High] Ability to make groups of tabs, that are not open, but will be open when you click to open the group.

You can open several notes in a split and hide/unhide that group in the latest version (v0.4.0). Or you can automatically hide all other groups and focus on the working note with one click (Zen Mode).

In the near future, it will be able to save/restore tabs and groups, browse layout history, or stash a group into a temporary inbox and retrieve them.

**Comment by oxdc** [comment]
> [HIGH] Add some keywords into your description for the plugin, for more people to find the plugin. Such as "Tabs Organizing" or "Tab Groups"

Done

> [High] Allow users to name grouped tab sections

Done

**Comment by Bluemandan** [comment]
Wow that was fast! Thanks for taking some of my suggestions! To further elaborate:

_**[Low] Top Bar reflects all appearance changes made**_
So for the customization aspect regarding color and icons, it would be nice to see it reflect on the horizontal tabs themselves. Also seeing the other tab groups that are closed on the horizontal bar would be cool and would be very accessible, just like Googles Group Tabs. Though this would take up more tab space, which is the whole goal of this entire plug-in.
![image](https://github.com/user-attachments/assets/738f4bf1-36cb-4070-a4f0-45a1ca5234ed)

**_[High] Ability to make groups of tabs, that are not open, but will be open when you click to open the group._**
Knowing that the restore tabs is somewhere in the future is great to hear! My only issue with your current version is the tab resizing. At any point I will only have 2 groups of tabs open, usually I will size them to fill the space. When I open another group of tabs, some times it will be displayed how I prefer it, split in half, equally or close to it, but sometimes it will get pinched on the side like in the image. Or if I am trying to resize the tab group, it will act like I am trying to reel in a fish, Ill pull to stretch the tab out, but then it will recline part of the way, then ill drag again, but it will move slight back. I don't have the solution to it other than a tab auto-sizing or adjusting it. 
![image](https://github.com/user-attachments/assets/712b3aa9-accf-4268-b3e6-e2aa465dab1b)

Otherwise everything is great! Thank you for commenting and considering everything. Excited to see your future implementations!

**Comment by oxdc** [comment]
> it would be nice to see it reflect on the horizontal tabs themselves
> seeing the other tab groups that are closed on the horizontal bar

Thanks for your explanation! These two features, although being useful, require hijacking the native tabs of Obsidian, which unfortunately might cause serious issues. I’ll look into it after refactoring and refining the drag-and-drop function. If I could make it work *safely*, this would be an optional feature. 🎈

> seeing the other tab groups that are closed on the horizontal bar

This might be an anti-pattern for tab management. Just my thought. Obsidian, unlike web browsers, supports splitting with multiple tab groups on screen. So, the question comes: where to put these “closed tab indicators”? I have not yet figured out the best solution. If you have any suggestions, it would be very helpful!

**Comment by Bluemandan** [comment]
Hmm, Ill get back to you on that, thought UI layouts are not my strongest suit, ill figure something out.

**Comment by oxdc** [comment]
> When I open another group of tabs, some times it will be displayed how I prefer it, split in half, equally or close to it, but sometimes it will get pinched on the side like so.

Interesting 🤔  Do you have the same issue without enabling *Vertical Tabs*? You might want to reopen Obsidian before testing. Since *Vertical Tabs* currently does not manipulate layouts, it seems to be an issue of Obsidian. Hiding is entirely achieved with CSS. Could you please post a screenshot **with all groups visible**? It would be helpful to see the full layout.

It seems that sometimes you want to maximize the active group (or the active note, really), and sometimes split the screen in half. If this is your use case, you do not have to repeatedly resize the group. Just click the Zen Mode button (the last one on the top), it will do this resizing for you!

**Comment by LionelVanArk** [comment]
Thank you for vertical tabs. I have two suggestions:

Request: I would like to select a hotkey modifier like "alt" for the setting "Enhanced Keyboard tab switching". Preferrable residing in the hotkeys pane settings.

Problem: Visually the tabs are only being numbered when this setting is turned on and you are pressing the ctrl key for a little time. This visual appears with a small delay. 

Suggestions: **The visuals settings should be standing on its own.**
- A switch to turn visual numbering on/off.
- Autohide visual numbering.
- Make it also appear with a different modifier key ("alt").
- Maybe Adjusting the delay the visuals appear with a slider.

The real solution can be different from my suggestions of course but overall i would appreciate a change towards these matters.

Regards

**Reply by oxdc** [reply]
Hi @LionelVanArk ! I've moved your suggestions to a new dedicated issue. I'll close this old cluttered thread now.
